drivers/SmartThings/sonos/src/api/sonos_connection.lua

```diff
@@ -425,7 +468,27 @@ function SonosConnection:start()
     _open_coordinator_socket(self, household_id, player_id)
   end
 
-  self:refresh_subscriptions()
+
+  local reply_tx, reply_rx = cosock.channel.new()
+
+  self:refresh_subscriptions(reply_tx)
+
+  local reply = reply_rx:receive()
```

Comment:
From the docs, the WSS connection itself won't actually be refused if we're missing credentials. Instead, the first command we try to send (which is a `subscribe` command in our case) will let us know if the auth fails.

So we've added this reply channel API to be able to make the initial subscription establishment in the `:start()` method wait until we know that the subscription is allowed.